File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        
            unordered_map<char,char>gzip;   
            unordered_map<char,char>used;
        
            for(int i=0;i<s.size();i++){
                
                     if((gzip.find(s[i])!=gzip.end() and gzip[s[i]]!=t[i])){
                                    return false;
                     }   
                    
                    if(used.find(t[i])!=gzip.end() and used[t[i]]!=s[i]) return false;
                
                    gzip[s[i]]=t[i];
                    used[t[i]]=s[i];
                
            }
        
            return true;
        
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    
    
    def isIsomorphic(self, s: str, t: str) -> bool:
    
            dict1={}
            dict2={}
            
            for item1,item2 in zip(s,t):
                
                if item1 not in dict1:
                    dict1[item1]=item2
                else:
                    if dict1[item1]!=item2:
                            return False
                        
                if  item2 not in dict2:
                    dict2[item2]=item1
                else:
                    if dict2[item2]!=item1:
                            return False
                        
            
            return True
        